libdnf/rpm/solv/map_iterator.hpp

```diff
@@ -38,25 +39,25 @@ class SolvMap;
 
 class SolvMapIterator {
 public:
-    explicit SolvMapIterator(const Map * map);
-    SolvMapIterator(const SolvMapIterator & other) = default;
+    explicit SolvMapIterator(const Map * map) noexcept;
+    SolvMapIterator(const SolvMapIterator & other) noexcept = default;
 
     using iterator_category = std::forward_iterator_tag;
-    using difference_type = PackageId;
+    using difference_type = std::ptrdiff_t;
```

Comment:
I am curious why `std::ptrdiff_t` and not `signed integer` should be used there?

Reply:
Please could you add also a description to commit message? `Fix difference_type` in commit message is not informative much.

Reply:
Citation:
"std::ptrdiff_t is used for pointer arithmetic and array indexing, if negative values are possible. Programs that use other types, such as int, may fail on, e.g. 64-bit systems when the index exceeds INT_MAX or if it relies on 32-bit modular arithmetic."

libsolv Map uses `int` type for its size. And each byte represents 8 packages. `int` type is small for the `difference_type`.
However, libsolv `Id` type is internaly declared as `int` too. -> We can't make package with `id` > `INT_MAX` -> We can't set bit on position > `INT_MAX` in the map. -> **`int` type is sufficient for `SolvMapIterator::difference_type`** .

Should I change it to `int`?

Reply:
No please do not change it. It is not a problem, and because it must be like that for reldeplist, it would be nice to use the same approach.